CalibMuon/DTCalibration/plugins/DTTTrigResidualCorrection.cc

```diff
@@ -83,9 +86,22 @@ namespace dtCalibration {
     tTrigMap_ = &*tTrig;
 
     // Get vDrift record
-    ESHandle<DTMtime> mTimeHandle;
-    setup.get<DTMtimeRcd>().get(mTimeHandle);
-    mTimeMap_ = &*mTimeHandle;
+    if (readLegacyVDriftDB) {
+      ESHandle<DTMtime> mTimeHandle;
+      setup.get<DTMtimeRcd>().get(mTimeHandle);
```

Comment:
NOTE: this way of accessing data from the EventSetup is deprecated. We are in the midst of requiring all modules to first announce they are _consuming_ the data (via call to `esConsumes` in the constructor) and then use an `edm::ESGetToken<>` when getting the data.

Reply:
This has been already discussed above, see:
https://github.com/cms-sw/cmssw/pull/34612#pullrequestreview-714232119
in short, the migration here would require some refactoring in the package since this class is an helper that is called by the actual EDProducer. While this is definitely to be done, it is outside the scope of this PR and we agreed to implement it separately, see the review discussion above.